### PR TITLE
[tech debt] remove feature phone support

### PR DESCRIPTION
**Pull request by snopoke**
Jira: https://dimagi-dev.atlassian.net/browse/USH-1839
## Product Description
This feature has been deprecated for some time and is no longer in use:

Last J2ME form received on prod was 2018-12-13T01:10:31.189708Z

Most recent J2ME enabled builds:
* 2.28.0: 2016-07-08
* 2.28.1: 2016-07-14
* 2.28.4: 2016-08-02
* 2.27.9: 2017-03-11

I have not been able to determine when the last successful download of a JAD/JAR build was. I have looked 6 months back in logs and not found any successful downloads on prod or india.

## Technical Summary
Delete views and related code for producing and downloading CommCare JAD & JAR files.

### Safety story
Removes unsupported feature.

### Automated test coverage
Removing code

### QA Plan
None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by dannyroberts — APPROVED** [review]
Nice. I think with this we can also probably delete the ~entire `builds` app. Happy to take a look at that

File: corehq/apps/app_manager/templates/app_manager/partials/releases/releases_deploy_modal.html [hunk 1]
```diff
@@ -196,51 +196,6 @@ <h4 class="panel-title">
...
-              <div class="panel-body">
-                <ol data-bind="if: reveal_java_download()">
-                  <li>
-                    {% trans "Download <strong>both</strong> of the following files:" %}
-                    <p><a data-bind="attr: {href: $root.reverse('download_jad', id)}">{% trans "Download" %} CommCare.jad</a></p>
-                    <p><a data-bind="attr: {href: $root.reverse('download_jar', id)}">{% trans "Download" %} CommCare.jar</a></p>
-                  </li>
-                  <li>
-                    {% trans "For help on how to install, see" %}
-                    <a target="_blank" href="https://help.commcarehq.org/display/commcarepublic/CommCare+for+Feature+Phones">{% trans "Set Up Feature Phones" %}</a>.
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/releases/releases_deploy_modal.html:227]: It'd be good to remove this page and related public help content. Do you think the support team could take that?

**Review by orangejenny — APPROVED** [review]
hooray

File: corehq/apps/app_manager/templates/app_manager/partials/releases/releases_deploy_modal.html (hunk 1 above)
Comment (snopoke) [reply corehq/apps/app_manager/templates/app_manager/partials/releases/releases_deploy_modal.html:227]: Support has already been given the go-ahead to archive that page.

File: corehq/apps/app_manager/views/download.py [hunk 2]
```diff
@@ -445,7 +375,6 @@ def download_index(request, domain, app_id):
     return render(request, "app_manager/download_index.html", {
         'app': request.app,
         'files': OrderedDict(sorted(files.items(), key=lambda x: x[0] or '')),
-        'supports_j2me': request.app.build_spec.supports_j2me(),
```
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/apps/app_manager/views/download.py:448]: I think it is but I wanted to keep the scope of this contained. I think anything that mentions `j2me` can be removed.

**Comment by snopoke** [comment]
Putting this on staging as a precaution.

**Comment by orangejenny** [comment]
@snopoke Do you think this has been on staging long enough, or is it going to get actively QAed? (I was writing some tests the other day that needed to mock this code, so I was reminded of it.)

**Comment by snopoke** [comment]
> @snopoke Do you think this has been on staging long enough, or is it going to get actively QAed? (I was writing some tests the other day that needed to mock this code, so I was reminded of it.)

Yes, I do. I mostly wanted to make sure it didn't break any app build workflows.

**Pull request merged**